E0322 needs to be updated to new format

From: src/test/compile-fail/coherence-impls-sized.rs

Error E0322 needs a span_label, updating it from:

```
error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> src/test/compile-fail/coherence-impls-sized.rs:24:1
   |
24 | impl Sized for TestE {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> src/test/compile-fail/coherence-impls-sized.rs:24:1
   |
24 | impl Sized for TestE {} //~ ERROR E0322
   | ^^^^^^^^^^^^^^^^^^^^^^^ impl of `Sized` not allowed
```

Bonus: shrink the span to only the Sized part:

```
error[E0322]: explicit impls for the `Sized` trait are not permitted
  --> src/test/compile-fail/coherence-impls-sized.rs:24:1
   |
24 | impl Sized for TestE {} //~ ERROR E0322
   |      ^^^^^ impl of `Sized` not allowed
```

--- a/src/librustc_typeck/coherence/orphan.rs
+++ b/src/librustc_typeck/coherence/orphan.rs
@@ -331,8 +331,10 @@ impl<'cx, 'tcx> OrphanChecker<'cx, 'tcx> {
 
                 // Disallow *all* explicit impls of `Sized` and `Unsize` for now.
                 if Some(trait_def_id) == self.tcx.lang_items.sized_trait() {
-                    span_err!(self.tcx.sess, item.span, E0322,
-                              "explicit impls for the `Sized` trait are not permitted");
+                    struct_span_err!(self.tcx.sess, item.span, E0322,
+                              "explicit impls for the `Sized` trait are not permitted")
+                        .span_label(item.span, &format!("impl of 'Sized' not allowed"))
+                        .emit();
                     return;
                 }
                 if Some(trait_def_id) == self.tcx.lang_items.unsize_trait() {

--- a/src/test/compile-fail/coherence-impls-sized.rs
+++ b/src/test/compile-fail/coherence-impls-sized.rs
@@ -22,12 +22,15 @@ struct NotSync;
 impl !Sync for NotSync {}
 
 impl Sized for TestE {} //~ ERROR E0322
+//~^ impl of 'Sized' not allowed
 
 impl Sized for MyType {} //~ ERROR E0322
+//~^ impl of 'Sized' not allowed
 
 impl Sized for (MyType, MyType) {} //~ ERROR E0117
 
 impl Sized for &'static NotSync {} //~ ERROR E0322
+//~^ impl of 'Sized' not allowed
 
 impl Sized for [MyType] {} //~ ERROR E0117
 

